File: BinarytoDecimal.php
<?php
function BinarytoDecimal($binaryNumber) {
	$value = strval($binaryNumber);
	$length = strlen($value);
	$output = 0;
	for ($position = 0; $position < $length; $position++) {
		$char = substr($value, $position, 1);
		if ($char == '1') {
			$output = intval($output) + intval(2**($length - 1 - $position));
		} else if ($char == '0') {
			
		} else {
			return "Make sure you only input binary values";
		}
	}
	return $output;
}

$prompt = readline("Binary Number: ");
echo BinarytoDecimal($prompt) . "\n";
?>

File: BinarytoDecimal.nim
# nim compile --run BinarytoDecimal.nim
import strutils
import math

proc BinarytoDecimal(binaryNumber: int): int =
    var value: cString = intToStr(binaryNumber)
    var length: int = len(value)
    var output: int = 0
    var two: int = 2
    for position in 0..length-1:
        var chr = value[position]
        if chr == '1':
            output += two ^ (length - 1 - position).int
        else:
            if chr == '0':
                continue
            else:
                echo "Make sure you only input binary values"
                quit(0)
    return output

write(stdout, "Binary Number: ")
echo BinarytoDecimal(parseInt(readLine(stdin)))
